src/rules/no-empty-fields.ts

```diff
@@ -0,0 +1,105 @@
+import { createRule } from "../createRule";
+import { isString, omit } from "../utils/predicates";
+import detectNewline from "detect-newline";
+import detectIndent from "detect-indent";
+
+export const rule = createRule({
+	create(context) {
+		const objectFields = [
+			"peerDependencies",
+			"scripts",
+			"dependencies",
+			"devDependencies",
+		];
+		const arrayFields = ["files"];
```

Comment:
[Refactor] There's no need for these to be re-created inside the `create(context) {` function each time. They should be moved to outside the `rule` definition.